examples/stokes/source/stokes_ida.cc

```diff
@@ -0,0 +1,858 @@
+  timer_outfile("timer.txt"),
+  tcout (timer_outfile,
+         (Utilities::MPI::this_mpi_process(comm)
+          == 0)),
+  computing_timer (comm,
+                   tcout,
+                   TimerOutput::summary,
+                   TimerOutput::wall_times),
+
+  eh("Error Tables", "u,u,p",
+     "L2,H1;AddUp;L2"),
+
+  pgg("Domain"),
+  fe_builder("Finite Element",
+             "FESystem[FE_Q(2)^dim-FE_Q(1)]",
+             "u,u,p"),
+
+  exact_solution("Exact solution"),
+  forcing_term("Forcing term"),
+  initial_solution("Initial solution"),
+  initial_solution_dot("Initial solution_dot"),
+  dirichlet_bcs("Dirichlet BCs", "u,u,p", "0=u"),
+
+  data_out("Output Parameters", "vtu"),
+  dae(*this)
+{}
+
+template <int dim>
+void Stokes<dim>::declare_parameters (ParameterHandler &prm)
+{
+  add_parameter(  prm,
+                  &initial_global_refinement,
+                  "Initial global refinement",
+                  "1",
+                  Patterns::Integer (0));
+
+  add_parameter(  prm,
+                  &max_time_iterations,
+                  "Maximum number of time steps",
+                  "10000",
+                  Patterns::Integer (0));
+
+  add_parameter(  prm,
+                  &timer_file_name,
+                  "Timer output file",
+                  "timer.txt",
+                  Patterns::FileName());
+
+
+  add_parameter(  prm,
+                  &adaptive_refinement,
+                  "Adaptive refinement",
+                  "true",
+                  Patterns::Bool());
+
+
+  add_parameter(  prm,
+                  &use_direct_solver,
+                  "Use direct solver if available",
+                  "true",
+                  Patterns::Bool());
+
+  add_parameter(  prm,
+                  &use_space_adaptivity,
+                  "Refine mesh during transient",
+                  "true",
+                  Patterns::Bool());
+
+  add_parameter(  prm,
+                  &kelly_threshold,
+                  "Threshold for restart solver",
+                  "1e-2",
+                  Patterns::Double(0.0));
+
+  add_parameter(  prm,
+                  &max_cells,
+                  "Maximum number of cells",
+                  "1000",
+                  Patterns::Integer(),
+                  "If negative, there is no upper bound");
+
+  add_parameter(  prm,
+                  &top_fraction,
+                  "Top fraction",
+                  "0.3",
+                  Patterns::Double(0.0));
+
+  add_parameter(  prm,
+                  &bottom_fraction,
+                  "Bottom fraction",
+                  "0.1",
+                  Patterns::Double(0.0));
+
+  add_parameter(  prm,
+                  &mu,
+                  "mu",
+                  "1.",
+                  Patterns::Double(0.0));
+}
+
+template <int dim>
+void Stokes<dim>::make_grid_fe()
+{
+  triangulation = SP(pgg.distributed(comm));
+  dof_handler = SP(new DoFHandler<dim>(*triangulation));
+  fe=SP(fe_builder());
```

Comment:
If we want the cylindrical hole we need to set up the manifold here. The only way I know is with something like this

const Point<2> center (0,0);
const double inner_radius = 0.2;

  const SphericalManifold<2> manifold_description(center);
  Triangulation<2>::active_cell_iterator
  cell = triangulation.begin_active(),
  endc = triangulation.end();
  for (; cell!=endc; ++cell)
    {
      for (unsigned int v=0;
           v < GeometryInfo<2>::faces_per_cell;
           ++v)
        {
               const double distance_from_center
        = center.distance (cell->face(v)->center());
        if (std::fabs(distance_from_center) < inner_radius + 1e-6)
          {
            cell->face(v)->set_manifold_id (99);
            break;
}}
triangulation.set_manifold (99, manifold_description);